teps/0040-capture-step-exit-code.md

```diff
@@ -0,0 +1,70 @@
+---
+status: proposed
+title: 'Capture Step Exit Code'
+creation-date: '2021-01-06'
+last-updated: '2021-01-06'
+authors:
+- '@pritidesai'
+- '@afrittoli'
+---
+
+# TEP-0040: Capture Step Exit Code
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Tekton tasks are defined as a collection of steps in which each step can specify a container image to run.
+Steps are executed in order in which they are specified. One single step failure results in a task failure
+in turn results in a pipeline failure. Once a step results in failure, rest of the steps are not executed.
+
+Many common pipelines have requirement where a step failure must not halt the entire execution. In order to
+continue executing a task even if a step fail, pipeline authors have flexibility of wrapping an image and
+exiting that task with success. This changes the failing task into success and does not block further
+execution. This is a workaround and only works with images which can be wrapped. This workaround does not apply
+to off-the-shelf container images.
+
+As a pipeline execution engine, we want to support off-the-shelf container image as a step and provide
+an option to capture non-zero exit code. The pipeline author can choose to continue execution after capturing
+the non-zero exit code and make it available to access it after the execution.
+
+Issue: [tektoncd/pipeline#2800](https://github.com/tektoncd/pipeline/issues/2800)
```

Comment:
I think this might also be covering https://github.com/tektoncd/pipeline/issues/1559 ?

Reply:
Indeed, even though the approach there is slightly different. The focus on that one is to let a step run even though there was a previous failure. Here is on letting the potentially failing step declare that the failure should be captured, so other steps don't need to be aware about the failure at all.